AFQ/data.py

```diff
@@ -2228,7 +2223,7 @@ def write_json(fname, data):
     None
     """
     with open(fname, 'w') as ff:
-        json.dump(data, ff)
+        json.dump(data, ff, default=lambda obj: "Not Serializable")
```

Comment:
What's the purpose of this? Wouldn't this cause some silent failures where no data would get written, but we'd just move on assuming it was written?

Reply:
`If specified, default should be a function that gets called for objects that can’t otherwise be serialized. It should return a JSON encodable version of the object or raise a TypeError. If not specified, TypeError is raised.`

When we write our metadata, sometimes what we try to write, like the seed_mask, are objects that are not always serializable. So the file still gets written, its just that non-serializable objects will be printed as "Not Serializable" instead of raising a TypeError.

Reply:
Actually, seed_mask was not the problem. I think that passing a RandomState() object to Segmentation is a better example. We will try to serialize that object when we write the metadata. Previously, it would crash. Now, it simply prints something like this: "rng: Not Serializable" in the long list of other parameters.